### PR TITLE
Added nw-chromedriver-installer module

### PR DESCRIPTION
**Pull request by kierans**
Can I get a review of this @insightfuls please? Looking for any obvious bugs, pitfalls, etc.

### CODE REVIEW

**Review by insightfuls — CHANGES_REQUESTED** [review]
I think your versioning here has defeated your separation of concerns, because unless I've read something wrongly, you're reading the version of Chromedriver to download from this module's package.json. That means you've tied the version of NW.js to this implementation; if you need to change keys, or whatever, you won't be able to download old versions because their package will use the old keys. I think this module should be semantically versioned in its own right (and preferably never have a major version upgrade do backward compatibility is always maintained).

**Comment by kierans** [comment]
> you're reading the version of Chromedriver to download from this module's package.json

No I'm not as I'm starting the search for the `package.json` to read from from the `node_modules` dir using [pkg-up](https://www.npmjs.com/package/pkg-up) (see `install.js:63`)

You can test this by cloning master

```
$ npm install
...
Downloading '0.12.2/chromedriver-mac-x64' ...
...
```

The reason I published the alpha of `nw-chromedriver-installer` was to test this scenario because using `npm link` wasn't doing it right because of all the symlinks.

Maybe I should document that section of the code better.

**Comment by insightfuls** [comment]
Yeah, I hadn't quite read it properly, but I'm still not seeing how the versioning is going to work in prod. Because when this is installed in practice it will be in `node_modules` alongside the `nw-chromedriver` module, because of the way `npm` flattens modules nowadays. That means `pkgup` is going to find the `package.json` of the project that is consuming `nw-chromedriver` and read its version, which is totally unrelated. I think you need a different strategy for figuring out the version.

**Comment by insightfuls** [comment]
Also, suggestion/enhancement request: could the downloaded binary go in a central cache (i.e. in a dot-directory in the user's home directory) so that if you install the module in multiple projects or blow away your `node_modules` it doesn't have to download the binary again?

Hoping we can get a similar change to `nwjs-builder-phoenix` as well.

**Comment by kierans** [comment]
Re enhancement request please open an issue for it 😄

File: installer/nw-chromedriver-version.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    .then(function(json) {
+      return json.version;
+    });
+};
+
+function findNwChromeDriver() {
+  const searchPath = path.resolve(path.join(__dirname, ".."));
+
+  return new Promise(function(resolve, reject) {
+    const readdir = function(searchPath) {
```
Comment (insightfuls) [inline installer/nw-chromedriver-version.js]: I don't mind all that much, as it's fairly localised, but perhaps consider whether there is too much shadowing going on here, with `fs.readdir` and `readdir`, and `searchPath` in two different scopes. Potential confusion for future self?

**Review by insightfuls — APPROVED** [review]
This algorithm seems better to me. It might die during development or do something strange if your checkout is not named `nw-chromedriver`, especially if a sibling checkout IS named that, but that's a strange edge case, and catering for it would probably make the algorithm strange or less efficient, so I think it's good as is.

Just a call-out of a possibly confusing bit of code you might like to refactor.

**Pull request merged**